contract/AElf.Contracts.Election/ElectionContract_Elector.cs

```diff
@@ -0,0 +1,238 @@
+using AElf.Contracts.MultiToken.Messages;
+using AElf.Contracts.Profit;
+using AElf.Contracts.Vote;
+using AElf.Sdk.CSharp;
+using AElf.Types;
+using Google.Protobuf;
+using Google.Protobuf.WellKnownTypes;
+
+namespace AElf.Contracts.Election
+{
+    /// <summary>
+    /// Vote & Withdraw
+    /// </summary>
+    public partial class ElectionContract
+    {
+        #region Vote
+
+        /// <summary>
+        /// Call the Vote function of VoteContract to do a voting.
+        /// </summary>
+        /// <param name="input"></param>
+        /// <returns></returns>
+        public override Empty Vote(VoteMinerInput input)
+        {
+            // Check candidate information map instead of candidates. 
+            var targetInformation = State.CandidateInformationMap[input.CandidatePubkey];
+            Assert(targetInformation != null, "Candidate not found.");
+            if (targetInformation == null) return new Empty(); // Just to avoid IDE warning.
+            Assert(targetInformation.IsCurrentCandidate, "Candidate quited election.");
+
+            var lockSeconds = (input.EndTimestamp - Context.CurrentBlockTime).Seconds;
+            Assert(lockSeconds >= State.MinimumLockTime.Value,
+                $"Invalid lock time. At least {State.MinimumLockTime.Value.Div(60).Div(60).Div(24)} days");
+            Assert(lockSeconds <= State.MaximumLockTime.Value,
+                $"Invalid lock time. At most {State.MaximumLockTime.Value.Div(60).Div(60).Div(24)} days");
+
+            State.LockTimeMap[Context.TransactionId] = lockSeconds;
+
+            var recoveredPublicKey = Context.RecoverPublicKey();
+
+            UpdateElectorInformation(recoveredPublicKey, input.Amount);
+
+            UpdateCandidateInformation(input.CandidatePubkey, input.Amount);
+
+            State.TokenContract.Lock.Send(new LockInput
+            {
+                Address = Context.Sender,
+                Symbol = Context.Variables.NativeSymbol,
+                LockId = Context.TransactionId,
+                Amount = GetElfAmount(input.Amount),
+                Usage = "Voting for Main Chain Miner Election."
+            });
+
+            State.TokenContract.Issue.Send(new IssueInput
+            {
+                Symbol = ElectionContractConstants.VoteSymbol,
+                To = Context.Sender,
+                Amount = input.Amount,
+                Memo = "Issue VOTEs."
+            });
+
+            State.VoteContract.Vote.Send(new VoteInput
```

Comment:
Locking tokens and maintaining each candidates vote amounts (in UpdateCandidateInformation) are maintained in Election contract. Why do we still need to rely on vote contract?